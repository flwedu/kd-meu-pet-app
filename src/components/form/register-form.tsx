import { FormEvent, useState } from "react";
import { RequiredTextField } from "../form-components/required-text-field";

export function RegisterForm() {
  const [data, setData] = useState({ username: "", password: "" });

  function handleChange(e: FormEvent<HTMLInputElement>) {
    const { name, value } = e.currentTarget;

    setData({ ...data, [name]: value });
  }

  function handleSubmit() {
    console.log("Submitted");
  }

  return (
    <form
      className="form form-login align-center flex flex-col rounded"
      onSubmit={handleSubmit}
    >
      <div onChange={handleChange}>
        <h2>Formul√°rio de cadastro</h2>
        <p>Por favor, insira seus dados abaixo</p>
        <RequiredTextField
          name="name"
          placeholder="Nome completo"
          type="text"
          onChange={handleChange}
        />
        <RequiredTextField
          name="username"
          placeholder="Username"
          type="text"
          onChange={handleChange}
        />
        <RequiredTextField
          name="email"
          placeholder="E-mail"
          type="email"
          onChange={handleChange}
        />
        <RequiredTextField
          name="password"
          placeholder="Senha"
          type="password"
          onChange={handleChange}
        />
        <RequiredTextField
          name="password2"
          placeholder="Repita a senha"
          type="password"
          onChange={handleChange}
        />
      </div>
    </form>
  );
}
