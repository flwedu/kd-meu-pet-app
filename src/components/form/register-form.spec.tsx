import { fireEvent, render } from "@testing-library/react";
import { RegisterForm } from "./register-form";

describe("RegisterForm component tests", () => {
  const onSubmit = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });
  test("should render the component properly", () => {
    const { container } = render(<RegisterForm onSubmit={onSubmit} />);

    expect(container).toMatchSnapshot();
  });

  test("Should call onSubmit() only if all fields are valid", () => {
    const { getByPlaceholderText } = render(
      <RegisterForm onSubmit={onSubmit} />
    );

    const nameInput = getByPlaceholderText("Nome completo") as HTMLInputElement;
    const usernameInput = getByPlaceholderText("Username") as HTMLInputElement;
    const emailInput = getByPlaceholderText("E-mail") as HTMLInputElement;
    const passwordInput = getByPlaceholderText("Senha") as HTMLInputElement;
    const confirmPasswordInput = getByPlaceholderText(
      "Repita a senha"
    ) as HTMLInputElement;

    const form = nameInput.form!;
    fireEvent.submit(form);

    expect(onSubmit).not.toHaveBeenCalled();

    fireEvent.change(nameInput, { target: { value: "test" } });
    fireEvent.change(usernameInput, { target: { value: "test" } });
    fireEvent.change(emailInput, { target: { value: "testemail@email.com" } });
    fireEvent.change(passwordInput, { target: { value: "test" } });
    fireEvent.change(confirmPasswordInput, { target: { value: "test" } });

    fireEvent.submit(form);

    expect(onSubmit).toHaveBeenCalledTimes(1);
  });
});
