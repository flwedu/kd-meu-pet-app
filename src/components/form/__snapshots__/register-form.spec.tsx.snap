// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm component tests should render the component properly 1`] = `
<div>
  <form
    class="form form-login align-center flex flex-col rounded"
  >
    <div>
      <h2>
        Formul√°rio de cadastro
      </h2>
      <p>
        Por favor, insira seus dados abaixo
      </p>
      <div
        class="field"
      >
        <input
          class="input"
          name="name"
          placeholder="Nome completo"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="field"
      >
        <input
          class="input"
          name="username"
          placeholder="Username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="field"
      >
        <input
          class="input"
          name="email"
          placeholder="E-mail"
          required=""
          type="email"
          value=""
        />
      </div>
      <div
        class="field"
      >
        <input
          class="input"
          name="password"
          placeholder="Senha"
          required=""
          type="password"
          value=""
        />
      </div>
      <div
        class="field"
      >
        <input
          class="input"
          name="confirmPassword"
          placeholder="Repita a senha"
          required=""
          type="password"
          value=""
        />
      </div>
    </div>
  </form>
</div>
`;
